src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -689,3 +689,20 @@ def extensions(self):
             extensions.append(x509.Extension(oid, critical, value))
 
         return x509.Extensions(extensions)
+
+    def public_bytes(self, encoding):
+        if not isinstance(encoding, serialization.Encoding):
+            raise TypeError("encoding must be an item from the Encoding enum")
+
+        # TODO: make text prelude optional.
+        bio = self._backend._create_mem_bio()
+        if encoding is serialization.Encoding.PEM:
+            res = self._backend._lib.X509_REQ_print(bio, self._x509_req)
```

Comment:
Print represents a very OpenSSL specific behavior, which conflicts with our goal of having the API be backend agnostic. Additionally, all of that information is available via other means via the existing APIs (and in fact print has limitations that our APIs do not for outputting several extension types).

You can call print for debug purposes by doing something like `openssl req -text` or `openssl x509 -text` on the command line of course, but if you think there's value in debug for this we could discuss having private methods on the openssl backend that call the needed functions as a separate PR.